In [2]:
import pandas as pd
import datetime

## Считываем данные

In [3]:
df = pd.read_excel('data.xlsx', sheet_name='Справочник точек') 

/tmp/ipykernel_33060/4208502316.py:1: FutureWarning: Inferring datetime64[ns] from data containing strings is deprecated and will be removed in a future version. To retain the old behavior explicitly pass Series(data, dtype=datetime64[ns])
  df = pd.read_excel('data.xlsx', sheet_name='Справочник точек')


In [4]:
df_revenue = pd.read_excel('data.xlsx', sheet_name='Выручка по обучающей выборке')

In [5]:
df['id точки'] = df['id точки'].astype(float).astype('Int64')

In [6]:
df_revenue['id точки'] = df_revenue['id точки'].astype(float).astype('Int64')

In [7]:
for column in df_revenue.select_dtypes(include='object').columns:
    df_revenue[column] = df_revenue[column].apply(lambda x: float(str(x).replace(' ', '').replace(',', '.')))
    df_revenue[column] = df_revenue[column].astype(float)

In [8]:
df_revenue.head()

id точки     2015-01-01     2015-02-01     2015-03-01     2015-04-01  \
0     11366  245709.342667  301024.685000  406199.782258  295602.853000   
1     11810  270245.842066  297964.820199  406198.175918  232433.263996   
2     15041  171972.976000  198141.440000  260215.741935  205316.010644   
3     15775  409032.989736  468816.460949  582717.510920  442807.529886   
4     16102  437693.191667  510077.198214  602107.617742  465779.798333   

      2015-05-01     2015-06-01     2015-07-01     2015-08-01     2015-09-01  \
0  268179.172258  266579.009667  202667.039355  207347.022581  237393.216667   
1  229317.708387  251403.308057  242000.555806  260407.109247  267051.162996   
2  187059.870968  198590.970667  178250.637781  187094.699355  211000.005517   
3  371689.316054  428430.990000  365894.971255  384266.496739  385515.509997   
4  423448.914516  414476.575000  370944.770968  392089.754839  416797.330000   

   ...     2018-01-01     2018-02-01     2018-04-01     2018-05-01  \
0  ...  271084.023513  298845.267500  277776.347333  270902.919677   
1  ...  245402.383180  273515.012143  240088.292000  196260.466452   
2  ...  174154.623627  173190.240000  240396.837333  241494.616774   
3  ...  431909.251500  493571.089286  408575.490000  387081.406452   
4  ...  378429.858867  453511.489286  388849.670000  369223.219355   

      2018-06-01     2018-07-01     2018-08-01     2018-09-01     2018-10-01  \
0  280106.099000  243448.836129  236709.816452  248268.778333  296637.103226   
1  228920.589000  204058.825161  207104.013135  203499.409840  206013.017419   
2  264873.765333  227461.233548  249503.230968  251805.680000  263152.960000   
3  418398.840000  362395.248387  393423.329032  387866.910000  421387.374194   
4  392389.013333  328593.246774  350885.291935  369367.358333  427575.006452   

      2018-11-01  
0  314456.870000  
1  218636.436000  
2  269913.018667  
3  481330.143000  
4  472508.001667  

[5 rows x 47 columns]

In [8]:
df_revenue.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155 entries, 0 to 154
Data columns (total 47 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id точки    152 non-null    Int64  
 1   2015-01-01  116 non-null    float64
 2   2015-02-01  116 non-null    float64
 3   2015-03-01  116 non-null    float64
 4   2015-04-01  116 non-null    float64
 5   2015-05-01  117 non-null    float64
 6   2015-06-01  117 non-null    float64
 7   2015-07-01  117 non-null    float64
 8   2015-08-01  118 non-null    float64
 9   2015-09-01  119 non-null    float64
 10  2015-10-01  119 non-null    float64
 11  2015-11-01  120 non-null    float64
 12  2015-12-01  126 non-null    float64
 13  2016-01-01  127 non-null    float64
 14  2016-02-01  130 non-null    float64
 15  2016-03-01  131 non-null    float64
 16  2016-04-01  131 non-null    float64
 17  2016-05-01  131 non-null    float64
 18  2016-06-01  129 non-null    float64
 19  2016-07-01  132 non-null    f

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246 entries, 0 to 245
Data columns (total 21 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id точки             245 non-null    Int64         
 1   Адрес                246 non-null    object        
 2   Широта               242 non-null    float64       
 3   Долгота              242 non-null    float64       
 4   Дата открытия        245 non-null    object        
 5   Дата закрытия        9 non-null      datetime64[ns]
 6   Площадь магазина     246 non-null    int64         
 7   Тип входа            244 non-null    object        
 8   Площадь ТЦ           204 non-null    float64       
 9   Формат магазина      246 non-null    object        
 10  Парковка             226 non-null    object        
 11  Этаж                 241 non-null    object        
 12  Время работы         232 non-null    object        
 13  Заполненность ТЦ     182 non-null  

### Анализ "id точек"

In [10]:
df[df['id точки'].isna()]

id точки                                              Адрес     Широта  \
229      <NA>  Тульская обл, г. Тула, ул. Советская, дом № 47...  37.618866   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
229  54.191774  2010-03-23 00:00:00           NaT               189   

           Тип входа  Площадь ТЦ Формат магазина  ... Этаж      Время работы  \
229  сквозной проход     51000.0         Мини ТЦ  ...    1  с 10:00 до 22:00   

    Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
229              2.0                    0                0              1.0   

     Расположение в ТЦ     Толока       Люкс    Выборка  
229                2.0  49.039977  73.137931  Обучающая  

[1 rows x 21 columns]

Есть запись с отсутствующим id

In [11]:
df[df['id точки'].duplicated(keep=False)]

id точки                                              Адрес     Широта  \
186     42645  Респ. Татарстан, г. Нижнекамск, ул. Химиков, д...  51.814376   
194     42683  Россия, Пензенская область, Пенза, проспект Ст...  44.951652   
195     42683     Россия, Пермский край, Пермь, улица Попова, 16  56.228132   
238     42645  Респ. Татарстан, г. Нижнекамск, ул. Химиков, д...  51.814376   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
186  55.631285  2006-04-16 00:00:00           NaT               255   
194  53.220492  2013-12-27 00:00:00           NaT               578   
195  58.008433  2012-07-15 00:00:00           NaT               277   
238  55.631285  2006-04-16 00:00:00           NaT               255   

                 Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
186  без сквозного прохода      2887.0         Мини ТЦ  ...    1   
194  без сквозного прохода     54500.0              ТЦ  ...    1   
195  без сквозного прохода      6400.0         Мини ТЦ  ...    1   
238  без сквозного прохода      2887.0         Мини ТЦ  ...    1   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
186   с 9:00 до 21:00              3.0                    1                1   
194  с 10:00 до 22:00              3.0                    1                0   
195  с 10:00 до 21:00              NaN                    0                0   
238   с 9:00 до 21:00              3.0                    1                1   

     Конкуренты в ТЦ  Расположение в ТЦ     Толока       Люкс    Выборка  
186              NaN               10.0  29.981741  54.620690  Обучающая  
194              1.0                8.0  34.480876  60.500000  Обучающая  
195              NaN                8.0  33.196617  63.896552  Обучающая  
238              NaN               10.0  29.981741  54.620690  Обучающая  

[4 rows x 21 columns]

По "id точки" есть дубликаты. Для id 42645 записи полностью повторяются, для id 42683 различаются - нужно поменять id.

### Анализ "Адреса"

In [12]:
df[df['Адрес'].duplicated(keep=False)]

id точки                                              Адрес     Широта  \
76      42080  г. Санкт-Петербург, пл. Александра Невского, д...  30.386566   
77      42085  г. Санкт-Петербург, пл. Александра Невского, д...  30.386566   
186     42645  Респ. Татарстан, г. Нижнекамск, ул. Химиков, д...  51.814376   
238     42645  Респ. Татарстан, г. Нижнекамск, ул. Химиков, д...  51.814376   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
76   59.924573  2006-08-01 00:00:00           NaT               304   
77   59.924573  2007-06-20 00:00:00           NaT               313   
186  55.631285  2006-04-16 00:00:00           NaT               255   
238  55.631285  2006-04-16 00:00:00           NaT               255   

                  Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
76   отдельный вход с улицы     90000.0              ТЦ  ...    1   
77   отдельный вход с улицы     90000.0              ТЦ  ...    1   
186   без сквозного прохода      2887.0         Мини ТЦ  ...    1   
238   без сквозного прохода      2887.0         Мини ТЦ  ...    1   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
76   с 10:00 до 21:00              NaN                    0                1   
77   с 10:00 до 21:00              NaN                    0                1   
186   с 9:00 до 21:00              3.0                    1                1   
238   с 9:00 до 21:00              3.0                    1                1   

     Конкуренты в ТЦ  Расположение в ТЦ     Толока       Люкс    Выборка  
76               0.0               10.0  51.431866  67.137931  Обучающая  
77               0.0               10.0  51.431866  67.137931  Обучающая  
186              NaN               10.0  29.981741  54.620690  Обучающая  
238              NaN               10.0  29.981741  54.620690  Обучающая  

[4 rows x 21 columns]

Для точек с id 42080 и 42085 повторяется адрес, нужно уточнить являются ли это двумя точками по одному адресу или есть проблемы с данными.

### Анализ "Широты" и "Долготы"

In [13]:
df[(df['Широта'].isna()) | (df['Долгота'].isna())]

id точки                                              Адрес  Широта  \
148     42437  Московская область, г. Серпухов. ул. Советская...     NaN   
150     42443  Московская область, Пушкино, ул Красноармейско...     NaN   
167     42509                        Пермь, улица Революции 13к2     NaN   
189     42667  Республика Башкортостан, г.Уфа, ул. Рубежная, ...     NaN   

     Долгота        Дата открытия Дата закрытия  Площадь магазина  \
148      NaN  2017-05-23 00:00:00           NaT               341   
150      NaN  2016-11-16 00:00:00           NaT               661   
167      NaN  2018-04-10 00:00:00           NaT              1128   
189      NaN  2015-12-03 00:00:00           NaT               640   

           Тип входа  Площадь ТЦ Формат магазина  ... Этаж Время работы  \
148              NaN     21800.0              ТЦ  ...    1          NaN   
150              NaN     28000.0              ТЦ  ...    1          NaN   
167  сквозной проход     82394.0              ТЦ  ...    1          NaN   
189  сквозной проход     82394.0              ТЦ  ...    1          NaN   

    Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
148              NaN                    0                0              NaN   
150              NaN                    0                1              NaN   
167              3.0                    1                1              2.0   
189              NaN                    0                1              NaN   

     Расположение в ТЦ     Толока       Люкс    Выборка  
148                4.0  51.027292  62.758621   Тестовая  
150                2.0  21.857582  55.068966   Тестовая  
167               14.0  28.381511  64.034483  Обучающая  
189                1.0  49.403805  71.586207  Обучающая  

[4 rows x 21 columns]

Для точек с id 42437, 42443, 42509, 42667 отсутствуют значения "Широты" и "Долготы"

### Анализ "Даты открытия"

In [14]:
df[df['Дата открытия'].isna()]

id точки                                              Адрес      Широта  \
106     42220  Камчатский край, г. Петропавловск-Камчатский, ...  158.587359   

       Долгота Дата открытия Дата закрытия  Площадь магазина        Тип входа  \
106  53.071978           NaN           NaT              1488  сквозной проход   

     Площадь ТЦ Формат магазина  ...    Этаж      Время работы  \
106      9000.0              ТЦ  ...  цоколь  с 10:00 до 21:00   

    Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
106              NaN                    0                0              NaN   

     Расположение в ТЦ     Толока   Люкс    Выборка  
106                8.0  37.415914  63.12  Обучающая  

[1 rows x 21 columns]

Отсутствует дата открытия магазина для точки с id 42220.

### Анализ "Даты закрытия"

In [15]:
df[df['Дата закрытия'].notna()]

id точки                                              Адрес     Широта  \
29      18931     г. Москва, Калужское ш., 21 км, С1, ТРЦ "Мега"  37.491161   
85      42122  г. Санкт-Петербург, пр. Энгельса, дом № 120, Т...  30.321797   
89      42135  г. Санкт-Петербург, ул. Балканская, дом № 17, ...  30.380018   
92      42156  г. Санкт-Петербург, ул. Ефимова, дом № 2, Лите...  30.320827   
99      42192  г. Санкт-Петербург,ул. Ленсовета, дом № 101, Л...  30.348522   
136     42383  М.О. г. Мытищи, Шараповский проезд, 1, "ТЦ "Кр...  37.761646   
137     42388  М.О. г. Электросталь, пр. Ленина, 0/10, ТЦ "Эл...  38.437642   
173     42546  Респ. Башкортостан, Уфимский р-он, г. Уфа, ул....  55.940446   
200     42706  Россия, Свердловская область, Екатеринбург, ул...  60.538536   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
29   55.603834  2010-10-29 00:00:00    2018-03-18               425   
85   60.037216  2004-05-18 00:00:00    2017-04-10               112   
89   59.826235  2013-08-02 00:00:00    2018-03-09               443   
92   59.926354  2007-03-28 00:00:00    2018-04-07               338   
99   59.831734  2012-12-23 00:00:00    2017-08-01               321   
136  55.914736  2010-07-12 00:00:00    2017-07-22               236   
137  55.799296  2008-02-11 00:00:00    2017-10-23               377   
173  54.725815  2004-04-26 00:00:00    2018-01-11               276   
200  56.817178  2019-07-19 00:00:00    2017-12-31               446   

                 Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
29   без сквозного прохода    180500.0              ТЦ  ...    1   
85   без сквозного прохода     16000.0              ТЦ  ...    1   
89         сквозной проход     93000.0              ТЦ  ...    1   
92         сквозной проход     32200.0              ТЦ  ...    1   
99         сквозной проход         NaN         Мини ТЦ  ...    1   
136  без сквозного прохода      2000.0         Мини ТЦ  ...    1   
137  без сквозного прохода     60000.0              ТЦ  ...  NaN   
173  без сквозного прохода      5700.0              ТЦ  ...    1   
200  без сквозного прохода    113163.0              ТЦ  ...    1   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
29   c 10:00 до 24:00              3.0                    1                0   
85   с 10:00 до 22:00              NaN                    0                0   
89   с 10:00 до 22:00              3.0                    1                1   
92   с 10:00 до 22:00              2.0                    1                1   
99   с 10:00 до 22:00              3.0                    0                1   
136  с 10:00 до 22:00              3.0                    0                1   
137  с 10:00 до 22:00              3.0                    1                0   
173  с 10:00 до 21:00              NaN                    0                0   
200  с 10:00 до 22:00              3.0                    1                1   

     Конкуренты в ТЦ  Расположение в ТЦ     Толока        Люкс  Выборка  
29               NaN                9.0  91.022487  104.586207        -  
85               2.0                9.0  36.292908   48.137931        -  
89               NaN               11.0  32.486835   57.448276      NaN  
92               NaN                7.0  72.133192   68.034483        -  
99               NaN                9.0  33.427061   63.620690        -  
136              NaN                8.0  29.458197   63.551724        -  
137              NaN                7.0  40.731117   51.000000        -  
173              NaN                4.0  19.802614   53.724138        -  
200              NaN               11.0  22.870652   62.833333        -  

[9 rows x 21 columns]

Для точки с id 42706 дата закрытия раньше чем дата открытия

### Анализ "Площади магазина"

In [16]:
df[df['Площадь магазина'].isna()]

Empty DataFrame
Columns: [id точки, Адрес, Широта, Долгота, Дата открытия, Дата закрытия, Площадь магазина, Тип входа, Площадь ТЦ, Формат магазина, Парковка, Этаж, Время работы, Заполненность ТЦ, Рядом с эскалатором, Рядом со входом, Конкуренты в ТЦ, Расположение в ТЦ, Толока, Люкс, Выборка]
Index: []

[0 rows x 21 columns]

Нет значений с нулевыми значениями

In [17]:
df[df['Площадь ТЦ']<df['Площадь магазина']]

Empty DataFrame
Columns: [id точки, Адрес, Широта, Долгота, Дата открытия, Дата закрытия, Площадь магазина, Тип входа, Площадь ТЦ, Формат магазина, Парковка, Этаж, Время работы, Заполненность ТЦ, Рядом с эскалатором, Рядом со входом, Конкуренты в ТЦ, Расположение в ТЦ, Толока, Люкс, Выборка]
Index: []

[0 rows x 21 columns]

Нет значений когда площадь магазина больше площади ТЦ

### Анализ "Типа входа"

In [18]:
df[df['Тип входа'].isna()]

id точки                                              Адрес  Широта  \
148     42437  Московская область, г. Серпухов. ул. Советская...     NaN   
150     42443  Московская область, Пушкино, ул Красноармейско...     NaN   

     Долгота        Дата открытия Дата закрытия  Площадь магазина Тип входа  \
148      NaN  2017-05-23 00:00:00           NaT               341       NaN   
150      NaN  2016-11-16 00:00:00           NaT               661       NaN   

     Площадь ТЦ Формат магазина  ... Этаж Время работы Заполненность ТЦ  \
148     21800.0              ТЦ  ...    1          NaN              NaN   
150     28000.0              ТЦ  ...    1          NaN              NaN   

     Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  Расположение в ТЦ  \
148                    0                0              NaN                4.0   
150                    0                1              NaN                2.0   

        Толока       Люкс   Выборка  
148  51.027292  62.758621  Тестовая  
150  21.857582  55.068966  Тестовая  

[2 rows x 21 columns]

Отсутствует тип входа для точек с id 42437, 42443

In [19]:
df['Тип входа'].unique()

array(['без сквозного прохода', 'сквозной проход',
       'отдельный вход с улицы + сквозной проход',
       'отдельный вход с улицы', 'единственный вход в ТК', nan],
      dtype=object)

### Анализ "Формата магазина" и "Площади ТЦ"

In [20]:
df[df['Формат магазина'].isna()]

Empty DataFrame
Columns: [id точки, Адрес, Широта, Долгота, Дата открытия, Дата закрытия, Площадь магазина, Тип входа, Площадь ТЦ, Формат магазина, Парковка, Этаж, Время работы, Заполненность ТЦ, Рядом с эскалатором, Рядом со входом, Конкуренты в ТЦ, Расположение в ТЦ, Толока, Люкс, Выборка]
Index: []

[0 rows x 21 columns]

In [21]:
df['Формат магазина'].unique()

array(['ТЦ', 'Street', 'Мини ТЦ', 'Стрит', 'Мини-ТЦ', 'Strееt', 'Стрт'],
      dtype=object)

Являются ли Форматы 'Мини ТЦ' и 'Мини-ТЦ' одними и теми же форматами.

То же самое для 'Street', 'Стрит', 'Strееt' и 'Стрт'

In [22]:
p_format = (df['Формат магазина'] == 'ТЦ') | (df['Формат магазина'] == 'Мини ТЦ') | (df['Формат магазина'] == 'Мини-ТЦ')
df[(df['Площадь ТЦ'].isna()) & (p_format)]["id точки"]

15       17987
18       18331
20       18527
52       41958
82       42113
94       42163
99       42192
126      42319
133      42368
138      42389
141      42406
143      42419
144      42420
174      42548
181      42598
212      42745
227      42803
240    2633927
242    2633930
Name: id точки, dtype: Int64

Для точек с id 17987, 18331, 18527, 41958, 42113, 42163, 42192, 42319, 42368, 42389, 42406, 42419, 42420, 42548, 42598, 42745, 42803, 2633927, 2633930 отсутствуют значения "Площадь ТЦ"

In [23]:
p_format = (df['Формат магазина'] == 'Street') | (df['Формат магазина'] == 'Стрит') | (df['Формат магазина'] == 'Strееt') | (df['Формат магазина'] == 'Стрт')
df[(df['Площадь ТЦ'].notna()) & (p_format)]

id точки                                              Адрес     Широта  \
69     42051  г. Санкт-Петербург, Московский пр., дом № 205,...  30.320466   

      Долгота        Дата открытия Дата закрытия  Площадь магазина  \
69  59.845018  2006-10-30 00:00:00           NaT               173   

                 Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
69  отдельный вход с улицы     25000.0          Street  ...    1   

                                   Время работы Заполненность ТЦ  \
69  Пн-Сб с 10:00 до 21:00  Вс с 11:00 до 21:00              NaN   

    Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  Расположение в ТЦ  \
69                    0                1              0.0                NaN   

    Толока        Люкс    Выборка  
69    98.0  114.413793  Обучающая  

[1 rows x 21 columns]

Для точки с id 42051 указана площадь ТЦ, хотя формат магазина указан "Street"

### Анализ "Парковки"

In [24]:
ids = list(df[df['Парковка'].isna()]['id точки'])

In [25]:
print('Отсутствует информация о парковке для точек с id', ids)

Отсутствует информация о парковке для точек с id [18010, 42001, 42056, 42059, 42072, 42100, 42118, 42132, 42189, 42191, 42288, 42345, 42437, 42443, 42499, 42505, 42780, 42800, 42810, 2633928]


In [26]:
df['Парковка'].unique()

array(['бесплатная парковка', 100, nan, 'платная парковка',
       'нет парковки', 1000, 700, 'бесплатная паpковка'], dtype=object)

Следует объеденить 'бесплатная парковка' и 'бесплатная паpковка'

Для некоторых записей есть числовые значения парковки (видимо, подразумевается стоимость оплаты), а для некоторых просто указана 'платная парковка'. Возможно, стоит вынести стоимость парковки в отдельную столбец.

### Анализ "Этажа"

In [27]:
df[df['Этаж'].isna()]

id точки                                              Адрес     Широта  \
43      41914  г. Москва, ул. Б. Черемушкинская, дом № 1, ТРЦ...  37.602001   
137     42388  М.О. г. Электросталь, пр. Ленина, 0/10, ТЦ "Эл...  38.437642   
149     42438  Московская область, Мытищи, Шараповский проезд, 2  37.759030   
181     42598  Респ. Татарстан, г. Казань, пр. Ибрагимова, до...  49.093290   
219     42775  Свердловская обл, г. Нижний Тагил, ул. Ленина,...  59.959410   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
43   55.689265  2010-01-18 00:00:00           NaT               440   
137  55.799296  2008-02-11 00:00:00    2017-10-23               377   
149  55.916592  2013-12-26 00:00:00           NaT               339   
181  55.821254  2017-01-06 00:00:00           NaT               934   
219  57.908150  2010-02-25 00:00:00           NaT               286   

                  Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
43    без сквозного прохода     92000.0              ТЦ  ...  NaN   
137   без сквозного прохода     60000.0              ТЦ  ...  NaN   
149   без сквозного прохода     30000.0              ТЦ  ...  NaN   
181  единственный вход в ТК         NaN         Мини ТЦ  ...  NaN   
219   без сквозного прохода      7200.0              ТЦ  ...  NaN   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
43   с 10:00 до 22:00              3.0                    0                0   
137  с 10:00 до 22:00              3.0                    1                0   
149  с 10:00 до 22:00              3.0                    1                0   
181  с 10:00 до 22:00              3.0                    0                0   
219  с 10:00 до 21:00              3.0                    0                0   

     Конкуренты в ТЦ  Расположение в ТЦ     Толока       Люкс    Выборка  
43               1.0               11.0  20.958678  60.275862  Обучающая  
137              NaN                7.0  40.731117  51.000000          -  
149              NaN                9.0  52.552950  73.181818  Обучающая  
181              NaN               10.0  35.277917  70.379310  Обучающая  
219              NaN                9.0  52.499327  57.758621  Обучающая  

[5 rows x 21 columns]

Отсутствует информация об этаже для точек с id 41914, 42388, 42438, 42598, 42775

In [28]:
df['Этаж'].unique()

array([3, 1, 'цоколь', nan, 2, 4.5, 20,
       datetime.datetime(2019, 5, 2, 0, 0)], dtype=object)

Есть некорректные значения для колонки "Этаж" (4.5, 20, datetime.datetime(2019, 5, 2, 0, 0))

In [29]:
df[df['Этаж'] == 4.5]

id точки                                              Адрес    Широта  \
212     42745  Самарская обл, Самара г, Красноармейская ул, д...  50.12771   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
212  53.186504  2017-05-02 00:00:00           NaT              1058   

                 Тип входа  Площадь ТЦ Формат магазина  ... Этаж Время работы  \
212  без сквозного прохода         NaN              ТЦ  ...  4.5          NaN   

    Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
212              2.0                    1                1              NaN   

     Расположение в ТЦ    Толока       Люкс   Выборка  
212               13.0  56.23621  74.655172  Тестовая  

[1 rows x 21 columns]

Для id точки 42745, возможно, проблема  со считыванием данных и имеется в виду 4 и 5 этаж

In [30]:
df[df['Этаж'] == 20]

id точки                                              Адрес     Широта  \
232     42826  Тюменская обл, г. Тюмень, ул. Орджоникидзе, до...  65.541416   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
232  57.148674  2016-02-08 00:00:00           NaT               424   

           Тип входа  Площадь ТЦ Формат магазина  ... Этаж      Время работы  \
232  сквозной проход     20720.0              ТЦ  ...   20  с 10:00 до 21:00   

    Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
232              3.0                    0                0              0.0   

     Расположение в ТЦ     Толока        Люкс    Выборка  
232                6.0  59.699404  104.310345  Обучающая  

[1 rows x 21 columns]

Для id точки 42826, скорее всего опечатка, и точка не находится на 20 этаже

In [31]:
df[df['Этаж'] == datetime.datetime(2019, 5, 2, 0, 0)]

id точки                                              Адрес      Широта  \
234     42831  Хабаровский край, г. Хабаровск, ул. Ким-Ю-Чена...  135.082894   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
234  48.490178  2009-12-04 00:00:00           NaT               288   

           Тип входа  Площадь ТЦ Формат магазина  ...                 Этаж  \
234  сквозной проход     18000.0              ТЦ  ...  2019-05-02 00:00:00   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
234  с 10:00 до 22:00              3.0                    0                0   

     Конкуренты в ТЦ  Расположение в ТЦ      Толока       Люкс    Выборка  
234              NaN                8.0  102.390352  71.827586  Обучающая  

[1 rows x 21 columns]

Для id точки 42831, есть опечатка,указана дата вместо этажа

### Анализ "Заполненность ТЦ"

In [32]:
df['Заполненность ТЦ'].unique()

array([nan,  3.,  0.,  2.,  1.])

Что подразумевают "0", "1", "2", "3" под заполненностью ТЦ?

### Анализ "Рядом с эскалатором"

In [33]:
df[df['Рядом с эскалатором'].isna()]

Empty DataFrame
Columns: [id точки, Адрес, Широта, Долгота, Дата открытия, Дата закрытия, Площадь магазина, Тип входа, Площадь ТЦ, Формат магазина, Парковка, Этаж, Время работы, Заполненность ТЦ, Рядом с эскалатором, Рядом со входом, Конкуренты в ТЦ, Расположение в ТЦ, Толока, Люкс, Выборка]
Index: []

[0 rows x 21 columns]

In [34]:
df['Рядом с эскалатором'].unique()

array([0, 1, 2])

In [35]:
df[df['Рядом с эскалатором'] == 2]

id точки                                              Адрес     Широта  \
15     17987  Волгоградская обл, Волгоград г, Университетски...  44.434011   

      Долгота        Дата открытия Дата закрытия  Площадь магазина  \
15  48.637886  2016-10-31 00:00:00           NaT               993   

                Тип входа  Площадь ТЦ Формат магазина  ... Этаж Время работы  \
15  без сквозного прохода         NaN              ТЦ  ...    1          NaN   

   Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
15              2.0                    2                2              NaN   

    Расположение в ТЦ     Толока       Люкс    Выборка  
15               13.0  70.835287  91.862069  Обучающая  

[1 rows x 21 columns]

Что означают значения “0”, “1”, “2” для колонки "Рядом с эскалатором"? Является ли “0” - не рядом, а “1” - рядом? Если да, то есть опечатка для точки с id 17987, указано значение “2”

### Анализ "Рядом со входом"

In [36]:
df[df['Рядом со входом'].isna()]

Empty DataFrame
Columns: [id точки, Адрес, Широта, Долгота, Дата открытия, Дата закрытия, Площадь магазина, Тип входа, Площадь ТЦ, Формат магазина, Парковка, Этаж, Время работы, Заполненность ТЦ, Рядом с эскалатором, Рядом со входом, Конкуренты в ТЦ, Расположение в ТЦ, Толока, Люкс, Выборка]
Index: []

[0 rows x 21 columns]

In [37]:
df['Рядом со входом'].unique()

array([0, 1, 2, 3])

In [38]:
df[((df['Рядом со входом'] == 2) | (df['Рядом со входом'] == 3))]

id точки                                              Адрес     Широта  \
15      17987  Волгоградская обл, Волгоград г, Университетски...  44.434011   
114     42253  Краснодарский край, г. Краснодар, ул. Уральска...  39.050053   
226     42800  Тверская обл, Калининский р-он, г. Тверь, Твер...  35.907557   

       Долгота        Дата открытия Дата закрытия  Площадь магазина  \
15   48.637886  2016-10-31 00:00:00           NaT               993   
114  45.035407  2011-10-05 00:00:00           NaT               249   
226  56.858085  2011-11-02 00:00:00           NaT               387   

                  Тип входа  Площадь ТЦ Формат магазина  ... Этаж  \
15    без сквозного прохода         NaN              ТЦ  ...    1   
114         сквозной проход    178000.0              ТЦ  ...    1   
226  отдельный вход с улицы         NaN          Street  ...    1   

         Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
15                NaN              2.0                    2                2   
114  с 10:00 до 22:00              2.0                    1                3   
226  с 10:00 до 21:00              NaN                    1                2   

     Конкуренты в ТЦ  Расположение в ТЦ     Толока       Люкс    Выборка  
15               NaN               13.0  70.835287  91.862069  Обучающая  
114              NaN               10.0  66.159523  71.000000   Тестовая  
226              NaN                3.0  31.287526  66.000000  Обучающая  

[3 rows x 21 columns]

Что означают значения “0”, “1”, “2”, “3” для колонки "Рядом со входом"?

### Анализ "Конкуренты в ТЦ"

In [39]:
df['Конкуренты в ТЦ'].unique()

array([ 1., nan,  0.,  2.,  3.])

Что означают значения “0”, “1”, “2”, “3” для колонки "Конкуренты в ТЦ"?

### Анализ "Расположение в ТЦ"

In [40]:
df['Расположение в ТЦ'].unique()

array([ 7. ,  9. ,  6. , 10. ,  0. , 10.5,  8. ,  1.5,  4. , 13. ,  nan,
        3. , 11. ,  5. , 12. , 14. ,  2. ,  1. ,  7.5])

### Анализ Выборки

In [41]:
df[df['Выборка'].isna()]

id точки                                              Адрес     Широта  \
89     42135  г. Санкт-Петербург, ул. Балканская, дом № 17, ...  30.380018   

      Долгота        Дата открытия Дата закрытия  Площадь магазина  \
89  59.826235  2013-08-02 00:00:00    2018-03-09               443   

          Тип входа  Площадь ТЦ Формат магазина  ... Этаж      Время работы  \
89  сквозной проход     93000.0              ТЦ  ...    1  с 10:00 до 22:00   

   Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  Конкуренты в ТЦ  \
89              3.0                    1                1              NaN   

    Расположение в ТЦ     Толока       Люкс  Выборка  
89               11.0  32.486835  57.448276      NaN  

[1 rows x 21 columns]

In [42]:
df['Выборка'].unique()

array(['Обучающая', 'Тестовая', '-', nan, 'Тест', 'Обyчающая'],
      dtype=object)

Для некоторых записей стоит '-'

Стоит объеденить 'Тест' и 'Тестовая'

In [43]:
len(df[df['Выборка'] == 'Обучающая'])

156

In [44]:
df[df['Выборка'] == 'Обучающая'].head()

id точки                                              Адрес      Широта  \
0      8955   Оренбургская обл, Оренбург г, Шарлыкское ш, д...   55.118272   
1     11007  Алтайский край, г. Барнаул, ул. Малахова, дом ...   83.696799   
2     11366  Амурская область, Благовещенск, улица Мухина, 114  127.510357   
3     11810  Архангельская обл, г. Архангельск , Троицкий п...   40.528249   
6     15621  Архангельская обл, г. Северодвинск, ул. Ломоно...   39.785522   

     Долгота        Дата открытия Дата закрытия  Площадь магазина  \
0  51.843127  2016-11-09 00:00:00           NaT               478   
1  53.356954  2010-04-26 00:00:00           NaT               332   
2  50.280435  2017-08-25 00:00:00           NaT               356   
3  64.532316  2012-11-03 00:00:00           NaT               317   
6  64.550364  2003-11-09 00:00:00           NaT               178   

                                  Тип входа  Площадь ТЦ Формат магазина  ...  \
0                     без сквозного прохода    130000.0              ТЦ  ...   
1                           сквозной проход     31500.0              ТЦ  ...   
2                     без сквозного прохода     36000.0              ТЦ  ...   
3  отдельный вход с улицы + сквозной проход     35000.0              ТЦ  ...   
6                           сквозной проход     20000.0              ТЦ  ...   

  Этаж      Время работы Заполненность ТЦ  Рядом с эскалатором  \
0    3  с 10:00 до 22:00              NaN                    0   
1    1  с 10:00 до 22:00              NaN                    0   
2    1  с 10:00 до 22:00              3.0                    0   
3    1  с 10:00 до 21:00              3.0                    0   
6    1  с 10:00 до 21:00              3.0                    0   

   Рядом со входом  Конкуренты в ТЦ  Расположение в ТЦ      Толока       Люкс  \
0                0              1.0                7.0   37.857582  69.500000   
1                1              NaN                9.0   57.810494  66.896552   
2                1              1.0                6.0   49.959831  68.727273   
3                1              0.0               10.0  128.459158  64.517241   
6                0              0.0                8.0   82.880454  64.034483   

     Выборка  
0  Обучающая  
1  Обучающая  
2  Обучающая  
3  Обучающая  
6  Обучающая  

[5 rows x 21 columns]

In [45]:
len(df[df['Выборка'] == 'Тестовая'])

77

In [46]:
df[df['Выборка'] == 'Тестовая'].head()

id точки                                              Адрес     Широта  \
4      15041  Архангельская обл, г. Архангельск, Троицкий пр...  40.526317   
5      15087  Архангельская обл, г. Архангельск, ул. Воскрес...  40.535400   
9      16119  Башкортостан Респ, Уфа г, Цюрупы ул, дом № 97 ...  55.958263   
14     17983  Владимирская обл, г. Владимир, ул. Тракторная,...  40.387096   
21     18531  Воронежская обл, г. Воронеж, бул. Победы, дом ...  39.168108   

      Долгота        Дата открытия Дата закрытия  Площадь магазина  \
4   64.534327  2006-12-09 00:00:00           NaT               176   
5   64.541900  2014-11-20 00:00:00           NaT               416   
9   54.736556  2018-05-23 00:00:00           NaT               662   
14  56.149200  2008-06-27 00:00:00           NaT               227   
21  51.710399  2011-01-31 00:00:00           NaT               340   

                Тип входа  Площадь ТЦ Формат магазина  ...    Этаж  \
4         сквозной проход     15332.0              ТЦ  ...  цоколь   
5   без сквозного прохода     60000.0              ТЦ  ...       1   
9         сквозной проход     48000.0              ТЦ  ...       1   
14  без сквозного прохода     40000.0              ТЦ  ...       1   
21  без сквозного прохода     65000.0              ТЦ  ...       1   

        Время работы Заполненность ТЦ  Рядом с эскалатором  Рядом со входом  \
4   с 10:00 до 21:00              3.0                    1                0   
5   с 10:00 до 21:00              3.0                    1                1   
9   с 10:00 до 22:00              3.0                    1                1   
14  с 10:00 до 22:00              2.0                    1                0   
21  с 10:00 до 22:00              3.0                    1                1   

    Конкуренты в ТЦ  Расположение в ТЦ      Толока       Люкс   Выборка  
4               NaN                0.0  103.199885  78.275862  Тестовая  
5               NaN               10.5   71.418989  67.035714  Тестовая  
9               NaN               10.0   83.191620  71.428571  Тестовая  
14              NaN                4.0   55.689602  70.620690  Тестовая  
21              NaN               11.0   60.622333  66.000000  Тестовая  

[5 rows x 21 columns]

## Анализ Выручки по обучающей выборке

In [47]:
df_revenue

id точки     2015-01-01     2015-02-01     2015-03-01     2015-04-01  \
0       11366  245709.342667  301024.685000  406199.782258  295602.853000   
1       11810  270245.842066  297964.820199  406198.175918  232433.263996   
2       15041  171972.976000  198141.440000  260215.741935  205316.010644   
3       15775  409032.989736  468816.460949  582717.510920  442807.529886   
4       16102  437693.191667  510077.198214  602107.617742  465779.798333   
..        ...            ...            ...            ...            ...   
150   2634000  341009.722000  413489.085501  518426.278387  376071.505667   
151   2634008  270923.715000  404737.144643  470433.872222  401063.781000   
152   2634011  378429.753517  350607.477988  482925.193552  367225.245935   
153   2633936            NaN            NaN            NaN            NaN   
154      <NA>            NaN            NaN            NaN            NaN   

        2015-05-01     2015-06-01     2015-07-01     2015-08-01  \
0    268179.172258  266579.009667  202667.039355  207347.022581   
1    229317.708387  251403.308057  242000.555806  260407.109247   
2    187059.870968  198590.970667  178250.637781  187094.699355   
3    371689.316054  428430.990000  365894.971255  384266.496739   
4    423448.914516  414476.575000  370944.770968  392089.754839   
..             ...            ...            ...            ...   
150  357895.794516  505318.240505  458096.520286  516279.226774   
151  318507.450000  385579.165000  375166.269677  382299.377093   
152  349294.028363  388179.193333  400668.150126  421847.743068   
153  420226.940000  311592.339222  255945.113890  234998.024826   
154            NaN            NaN            NaN            NaN   

        2015-09-01  ...     2018-01-01     2018-02-01     2018-04-01  \
0    237393.216667  ...  271084.023513  298845.267500  277776.347333   
1    267051.162996  ...  245402.383180  273515.012143  240088.292000   
2    211000.005517  ...  174154.623627  173190.240000  240396.837333   
3    385515.509997  ...  431909.251500  493571.089286  408575.490000   
4    416797.330000  ...  378429.858867  453511.489286  388849.670000   
..             ...  ...            ...            ...            ...   
150  486665.455265  ...  513382.351353  590733.692857  480563.392000   
151  413219.906524  ...  162847.124247  202552.822857  185037.349333   
152  345782.485333  ...  461032.727239  443111.015436  410602.549333   
153  234465.065333  ...  360466.441020  372836.555000  299080.189333   
154            NaN  ...  502913.053140  594450.393471  560283.324000   

        2018-05-01     2018-06-01     2018-07-01     2018-08-01  \
0    270902.919677  280106.099000  243448.836129  236709.816452   
1    196260.466452  228920.589000  204058.825161  207104.013135   
2    241494.616774  264873.765333  227461.233548  249503.230968   
3    387081.406452  418398.840000  362395.248387  393423.329032   
4    369223.219355  392389.013333  328593.246774  350885.291935   
..             ...            ...            ...            ...   
150  457005.916452  541734.577000  444286.811935  574573.538710   
151  167999.870323  186120.868667  175067.582903  174118.710968   
152  379963.923226  457012.976000  431740.369032  491072.692000   
153  286674.025806  327415.778667  305542.250323  345389.650968   
154  514546.421768  644308.088000  533981.383006  562082.839742   

        2018-09-01     2018-10-01     2018-11-01  
0    248268.778333  296637.103226  314456.870000  
1    203499.409840  206013.017419  218636.436000  
2    251805.680000  263152.960000  269913.018667  
3    387866.910000  421387.374194  481330.143000  
4    369367.358333  427575.006452  472508.001667  
..             ...            ...            ...  
150  538086.317000  530747.271290  565599.127667  
151  174518.631333  189597.414516  226016.234333  
152  461658.858000  421671.512258  489829.227333  
153  349845.855333  361092.704994  363068.734000  
154  575194.942000  724658.394194  726229.02

In [48]:
df_revenue[df_revenue['id точки'].isna()]

id точки  2015-01-01  2015-02-01  2015-03-01  2015-04-01  2015-05-01  \
108      <NA>         NaN         NaN         NaN         NaN         NaN   
129      <NA>         NaN         NaN         NaN         NaN         NaN   
154      <NA>         NaN         NaN         NaN         NaN         NaN   

     2015-06-01  2015-07-01  2015-08-01  2015-09-01  ...     2018-01-01  \
108         NaN         NaN         NaN         NaN  ...            NaN   
129         NaN         NaN         NaN         NaN  ...  114585.758339   
154         NaN         NaN         NaN         NaN  ...  502913.053140   

        2018-02-01     2018-04-01     2018-05-01  2018-06-01     2018-07-01  \
108            NaN            NaN            NaN         NaN            NaN   
129  118510.400000   94791.263333   93654.952258  108397.950  100705.532258   
154  594450.393471  560283.324000  514546.421768  644308.088  533981.383006   

        2018-08-01     2018-09-01     2018-10-01  2018-11-01  
108            NaN            NaN            NaN         NaN  
129  116268.127419   97226.428333   97570.266129  121038.385  
154  562082.839742  575194.942000  724658.394194  726229.020  

[3 rows x 47 columns]

Есть записи с отсутствующими "id точек" (строки в excel 110, 131, 156)

In [49]:
df_revenue[df_revenue['id точки'].duplicated(keep=False)]

id точки     2015-01-01     2015-02-01    2015-03-01     2015-04-01  \
83      42383  227416.886667  301359.175000  3.664930e+05  256595.884697   
84      42383  394456.820997  454259.378600  5.836500e+05  443076.413799   
108      <NA>            NaN            NaN           NaN            NaN   
110     42597  159357.770000  163778.550000  2.092373e+05  119044.440667   
111     42597  788735.620309  915970.362829  1.215916e+06  872090.908507   
129      <NA>            NaN            NaN           NaN            NaN   
154      <NA>            NaN            NaN           NaN            NaN   

        2015-05-01     2015-06-01     2015-07-01     2015-08-01  \
83   223962.096129  241470.289667  268201.834194  285165.496452   
84   391290.346452  408554.289000  357205.572574  390696.346452   
108            NaN            NaN            NaN            NaN   
110  120611.429355  155005.578000  146695.845484  155073.011935   
111  780988.410697  822884.550667  697784.523255  757521.236129   
129            NaN            NaN            NaN            NaN   
154            NaN            NaN            NaN            NaN   

        2015-09-01  ...     2018-01-01     2018-02-01     2018-04-01  \
83   268671.797333  ...  290967.715455  310493.430000  264606.220667   
84   417790.097997  ...  482870.720640  508593.690000  438699.822000   
108            NaN  ...            NaN            NaN            NaN   
110  178703.464333  ...  258394.352932  303328.447500  232310.533000   
111  814036.689248  ...  849055.735613  953085.207143  787095.805333   
129            NaN  ...  114585.758339  118510.400000   94791.263333   
154            NaN  ...  502913.053140  594450.393471  560283.324000   

        2018-05-01     2018-06-01     2018-07-01     2018-08-01  \
83   219131.287742  279045.797333  239944.954516  294459.344065   
84   445378.677097  425878.134000  369380.528710  407637.960968   
108            NaN            NaN            NaN            NaN   
110  206991.946452  256464.936000  222763.039032  257462.160968   
111  804114.132903  817746.833333  665763.477419  735152.020645   
129   93654.952258  108397.950000  100705.532258  116268.127419   
154  514546.421768  644308.088000  533981.383006  562082.839742   

        2018-09-01     2018-10-01     2018-11-01  
83   270101.650000  238021.185161  287760.169667  
84   384249.393000  449429.469677  516463.893000  
108            NaN            NaN            NaN  
110  238893.624667  254131.913226  304643.126667  
111  704202.978667  828985.715355  945777.193333  
129   97226.428333   97570.266129  121038.385000  
154  575194.942000  724658.394194  726229.020000  

[7 rows x 47 columns]

Есть дубликаты с "id точки" 42383, 42597

In [50]:
df_revenue.head()

id точки     2015-01-01     2015-02-01     2015-03-01     2015-04-01  \
0     11366  245709.342667  301024.685000  406199.782258  295602.853000   
1     11810  270245.842066  297964.820199  406198.175918  232433.263996   
2     15041  171972.976000  198141.440000  260215.741935  205316.010644   
3     15775  409032.989736  468816.460949  582717.510920  442807.529886   
4     16102  437693.191667  510077.198214  602107.617742  465779.798333   

      2015-05-01     2015-06-01     2015-07-01     2015-08-01     2015-09-01  \
0  268179.172258  266579.009667  202667.039355  207347.022581  237393.216667   
1  229317.708387  251403.308057  242000.555806  260407.109247  267051.162996   
2  187059.870968  198590.970667  178250.637781  187094.699355  211000.005517   
3  371689.316054  428430.990000  365894.971255  384266.496739  385515.509997   
4  423448.914516  414476.575000  370944.770968  392089.754839  416797.330000   

   ...     2018-01-01     2018-02-01     2018-04-01     2018-05-01  \
0  ...  271084.023513  298845.267500  277776.347333  270902.919677   
1  ...  245402.383180  273515.012143  240088.292000  196260.466452   
2  ...  174154.623627  173190.240000  240396.837333  241494.616774   
3  ...  431909.251500  493571.089286  408575.490000  387081.406452   
4  ...  378429.858867  453511.489286  388849.670000  369223.219355   

      2018-06-01     2018-07-01     2018-08-01     2018-09-01     2018-10-01  \
0  280106.099000  243448.836129  236709.816452  248268.778333  296637.103226   
1  228920.589000  204058.825161  207104.013135  203499.409840  206013.017419   
2  264873.765333  227461.233548  249503.230968  251805.680000  263152.960000   
3  418398.840000  362395.248387  393423.329032  387866.910000  421387.374194   
4  392389.013333  328593.246774  350885.291935  369367.358333  427575.006452   

      2018-11-01  
0  314456.870000  
1  218636.436000  
2  269913.018667  
3  481330.143000  
4  472508.001667  

[5 rows x 47 columns]

In [51]:
all_data = df.merge(df_revenue, on='id точки', how='left')

In [52]:
all_data.head()

id точки                                              Адрес      Широта  \
0      8955   Оренбургская обл, Оренбург г, Шарлыкское ш, д...   55.118272   
1     11007  Алтайский край, г. Барнаул, ул. Малахова, дом ...   83.696799   
2     11366  Амурская область, Благовещенск, улица Мухина, 114  127.510357   
3     11810  Архангельская обл, г. Архангельск , Троицкий п...   40.528249   
4     15041  Архангельская обл, г. Архангельск, Троицкий пр...   40.526317   

     Долгота        Дата открытия Дата закрытия  Площадь магазина  \
0  51.843127  2016-11-09 00:00:00           NaT               478   
1  53.356954  2010-04-26 00:00:00           NaT               332   
2  50.280435  2017-08-25 00:00:00           NaT               356   
3  64.532316  2012-11-03 00:00:00           NaT               317   
4  64.534327  2006-12-09 00:00:00           NaT               176   

                                  Тип входа  Площадь ТЦ Формат магазина  ...  \
0                     без сквозного прохода    130000.0              ТЦ  ...   
1                           сквозной проход     31500.0              ТЦ  ...   
2                     без сквозного прохода     36000.0              ТЦ  ...   
3  отдельный вход с улицы + сквозной проход     35000.0              ТЦ  ...   
4                           сквозной проход     15332.0              ТЦ  ...   

      2018-01-01     2018-02-01     2018-04-01     2018-05-01     2018-06-01  \
0            NaN            NaN            NaN            NaN            NaN   
1            NaN            NaN            NaN            NaN            NaN   
2  271084.023513  298845.267500  277776.347333  270902.919677  280106.099000   
3  245402.383180  273515.012143  240088.292000  196260.466452  228920.589000   
4  174154.623627  173190.240000  240396.837333  241494.616774  264873.765333   

      2018-07-01     2018-08-01     2018-09-01     2018-10-01     2018-11-01  
0            NaN            NaN            NaN            NaN            NaN  
1            NaN            NaN            NaN            NaN            NaN  
2  243448.836129  236709.816452  248268.778333  296637.103226  314456.870000  
3  204058.825161  207104.013135  203499.409840  206013.017419  218636.436000  
4  227461.233548  249503.230968  251805.680000  263152.960000  269913.018667  

[5 rows x 67 columns]

In [53]:
points = list(df["id точки"])

In [54]:
revenue_points = list(df_revenue["id точки"])

In [55]:
print("В таблице 'Выручка по обучающей выборке' нет записей для точек с id:", list(set(points) - set(revenue_points)))

В таблице 'Выручка по обучающей выборке' нет записей для точек с id: [42505, 41995, 42009, 42013, 42546, 42039, 42553, 18497, 17987, 42567, 16971, 42059, 18010, 42595, 42085, 42598, 42089, 42090, 42618, 42113, 18568, 42122, 42128, 42133, 42645, 42663, 42155, 42667, 42668, 42161, 42163, 42176, 2633927, 2633928, 2633929, 2633930, 2633935, 42706, 42709, 42711, 2633946, 2633948, 42724, 42214, 15087, 16119, 42746, 8955, 42237, 11007, 42240, 15621, 42761, 42762, 42764, 42774, 42265, 42780, 42276, 42788, 42278, 42795, 18229, 42810, 42306, 42824, 42330, 18784, 42848, 41827, 42340, 41849, 42363, 41856, 41875, 42388, 42389, 18331, 42402, 41904, 42417, 42419, 42420, 16823, 41914, 41923, 42437, 41927, 42447, 18902, 41942, 42460, 42465, 42466, 41962, 42478, 42481, 42482, 18931]


## Вопросы

#### Сколько магазинов формата стрит в тестовой выборке?

In [56]:
test = ((df['Выборка'] == 'Тестовая') | (df['Выборка'] == 'Тест'))
street = ((df['Формат магазина'] == 'Street') | (df['Формат магазина'] == 'Стрит') | (df['Формат магазина'] == 'Strееt') | (df['Формат магазина'] == 'Стрт'))
print("Магазинов формата стрит в тестовой выборке", len(df[test & street]))

Магазинов формата стрит в тестовой выборке 7


#### Какова средняя выручка магазинов формата мини ТЦ за 2016 год?

In [57]:
all_data['2016'] = all_data[[x for x in all_data.columns if "2016" in x]].sum(axis=1)

In [58]:
mini = (all_data['Формат магазина'] == 'Мини ТЦ') | (all_data['Формат магазина'] == 'Мини-ТЦ')
print("Средняя выручка магазинов формата мини ТЦ за 2016 год:", all_data[mini]['2016'].mean())

Средняя выручка магазинов формата мини ТЦ за 2016 год: 2181378.443616589


#### Сколько магазинов с бесплатной парковкой?

In [59]:
free_parking = len(df[(df['Парковка'] == 'бесплатная парковка') | (df['Парковка'] == 'бесплатная паpковка')])
print("Магазинов с бесплатной парковкой", free_parking)

Магазинов с бесплатной парковкой 157
